encoding/ssz/query/ssz_interface.go

```diff
@@ -0,0 +1,24 @@
+package query
+
+import (
+	"errors"
+)
+
+type SSZObject interface {
+	HashTreeRoot() ([32]byte, error)
+}
+
```

Comment:
```suggestion
type SSZObject interface {
	ssz.Marshaler
    ssz.Unmarshaler
    ssz.HashRoot
}

```

where `ssz` is from `"github.com/prysmaticlabs/fastssz"`.

~~As `SSZObject` must be referring to a struct from `*.pb.go` file, I think inheriting all interfaces would be fine.~~ -> @rkapka pointed me out that we should have only needed methods, not inheriting all those interfaces. So just strikied through this line!

Reply:
Also I guess file name `ssz_interface.go` is not yet changed from previous commit.